packages/adapter-posthog/src/index.ts

```diff
@@ -0,0 +1,133 @@
+import { PostHog } from 'posthog-node';
+import type { PostHogAdapter, PostHogEntities, JsonType } from './types';
+
+export type { PostHogEntities, JsonType };
+
+export function createPostHogAdapter({
+  postHogKey,
+  postHogHost,
+}: {
+  postHogKey: string;
+  postHogHost: string;
+}): PostHogAdapter {
+  const client = new PostHog(postHogKey, {
+    host: postHogHost,
+  });
+
+  const result: PostHogAdapter = {
+    client,
+    isFeatureEnabled: (options) => {
+      return {
+        async decide({ key, entities, defaultValue }): Promise<boolean> {
+          const parsedEntities = parseEntities(entities);
+          const result =
+            (await client.isFeatureEnabled(
+              key,
+              parsedEntities.distinctId,
+              options,
+            )) ?? defaultValue;
+          if (result === undefined) {
```

Comment:
I'm wondering if we want to allow adapters to return undefined through the type definition. The core Flags SDK will already throw when a `decide` function returns undefined and no defaultValue was specified. Otherwise we will end up repeating these custom errors across all providers

Reply:
👍  I definitely only did this because I was fighting Typescript.